### ISSUE TITLE
Error with Addic7ed provider - list index out of range

### ISSUE THREAD

**Issue by Crupuk** [Diaoul/subliminal#522]
Is the addic7ed provider broken ? Can't find any open issue on this error : 

This is the log on a file with subtitle available on addic7ed website :

```
# /opt/venv/bin/subliminal --debug --addic7ed xxxxxx xxxxxx download -v -s -f -l fr -- 

INFO:subliminal.api:Listing subtitles with provider 'addic7ed' and languages set([<Language [fr]>])
INFO:subliminal.api:Initializing provider addic7ed
INFO:subliminal.providers.addic7ed:Logging in
DEBUG:subliminal.providers.addic7ed:Logged in
DEBUG:subliminal.providers.addic7ed:Getting show id
INFO:subliminal.providers.addic7ed:Getting the page of show id 30, season 12
ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
Traceback (most recent call last):
  File "/opt/venv/lib/python2.7/site-packages/subliminal/api.py", line 117, in list_subtitles
    provider_subtitles = self[name].list_subtitles(video, provider_languages)
  File "/opt/venv/lib/python2.7/site-packages/subliminal/providers/addic7ed.py", line 257, in list_subtitles
    return [s for s in self.query(video.series, video.season, video.year)
  File "/opt/venv/lib/python2.7/site-packages/subliminal/providers/addic7ed.py", line 226, in query
    match = series_year_re.match(soup.select('#header font')[0].text.strip()[:-10])
IndexError: list index out of range
```

Same error when no subtitle available :

```
# /opt/venv/bin/subliminal --debug --addic7ed xxxxxx xxxxxx download -v -s -f -l fr -- /dlna/shows/en/Greys\ Anatomy/S12/Greys.Anatomy.S12E02.720p.HDTV.x264-KILLERS.mkv

INFO:subliminal.api:Listing subtitles with provider 'addic7ed' and languages set([<Language [fr]>])
INFO:subliminal.api:Initializing provider addic7ed
INFO:subliminal.providers.addic7ed:Logging in
DEBUG:subliminal.providers.addic7ed:Logged in
DEBUG:subliminal.providers.addic7ed:Getting show id
INFO:subliminal.providers.addic7ed:Getting the page of show id 30, season 12
ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
Traceback (most recent call last):
  File "/opt/venv/lib/python2.7/site-packages/subliminal/api.py", line 117, in list_subtitles
    provider_subtitles = self[name].list_subtitles(video, provider_languages)
  File "/opt/venv/lib/python2.7/site-packages/subliminal/providers/addic7ed.py", line 257, in list_subtitles
    return [s for s in self.query(video.series, video.season, video.year)
  File "/opt/venv/lib/python2.7/site-packages/subliminal/providers/addic7ed.py", line 226, in query
    match = series_year_re.match(soup.select('#header font')[0].text.strip()[:-10])
IndexError: list index out of range
```

During my test i got another error :

```
# /opt/venv/bin/subliminal --debug --addic7ed xxxxxx xxxxxx download -v -s -f -l fr -- /dlna/shows/en/Greys\ Anatomy/S12/Greys.Anatomy.S12E01.REPACK.720p.HDTV.x264-KILLERS.mkv

INFO:subliminal.api:Listing subtitles with provider 'addic7ed' and languages set([<Language [fr]>])
INFO:subliminal.api:Initializing provider addic7ed
INFO:subliminal.providers.addic7ed:Logging in
DEBUG:subliminal.providers.addic7ed:Logged in
DEBUG:subliminal.providers.addic7ed:Getting show id
INFO:subliminal.providers.addic7ed:Getting the page of show id 30, season 12
ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
Traceback (most recent call last):
  File "/opt/venv/lib/python2.7/site-packages/subliminal/api.py", line 117, in list_subtitles
    provider_subtitles = self[name].list_subtitles(video, provider_languages)
  File "/opt/venv/lib/python2.7/site-packages/subliminal/providers/addic7ed.py", line 257, in list_subtitles
    return [s for s in self.query(video.series, video.season, video.year)
  File "/opt/venv/lib/python2.7/site-packages/subliminal/providers/addic7ed.py", line 227, in query
    series = match.group('series')
AttributeError: 'NoneType' object has no attribute 'group'
```

Thanks

**Comment by pannal** [comment]
The last one seems to be the same as https://github.com/pannal/Sub-Zero/issues/31
I got that when addic7ed decided I was querying too often. Maybe some request throttling would come in handy

**Comment by pannal** [comment]
Ah no, thats also punctuation.

**Comment by pannal** [comment]
Also the matching of the series name just before the download fails, it needs an updated regex to match the series' punctuation: ```series_year_re = re.compile('^(?P<series>[ \w.:\']+)(?: \((?P<year>\d{4})\))?$')```

**Comment by Diaoul** [comment]
Can you submit a PR with unittests?

**Comment by pannal** [comment]
Not for a change of 3 characters, sorry :)

**Comment by martinp** [comment]
Is this the same problem as #500?

**Comment by Diaoul** [comment]
I could reproduce this, it's due to Addic7ed returning 304s instead of the actual result. I don't know why and I couldn't easily fix it with basic cache control headers. I need to investigate this further.

**Comment by Diaoul** [comment]
Fixed with ea8b646c96e24649ec6fd436f5fb621e12d741ce and bc218fa13848182c167817eeba5bd81df719a7ac
